src/Cnab.Bundle/Parameter.cs

```diff
@@ -27,42 +50,82 @@ public class ParameterMetadata
         public string Description { get; set; }
     }
 
-    public class StringParameter : Parameter
+    public class StringParameter : Parameter<string>
     {
-        [JsonProperty("defaultValue")]
-        public string DefaultValue { get; set; }
-
-        [JsonProperty("allowedValues")]
-        public List<string> AllowedValues { get; set; }
-
         [JsonProperty("minLength")]
         public int MinimumLength { get; set; }
 
         [JsonProperty("maxLength")]
         public int MaximumLength { get; set; }
-    }
 
-    public class BoolParameter : Parameter
-    {
-        [JsonProperty("defaultValue")]
-        public bool DefaultValue { get; set; }
+        public override bool IsValid(string value)
+        {
+            if (!this.AllowedValues.Contains(value))
+                return false;
```

Comment:
Sorry but I think this is useless as feedback.  The consumer's _very first_ question is going to be _why_ the value is invalid and you give them no information about this.  Please change the design.